zebra-network/src/peer/handshake.rs

```diff
@@ -588,13 +588,29 @@ where
     debug!(?our_version, "sending initial version message");
     peer_conn.send(our_version).await?;
 
-    let remote_msg = peer_conn
+    let mut remote_msg = peer_conn
         .next()
         .await
         .ok_or(HandshakeError::ConnectionClosed)??;
 
-    // Check that we got a Version and destructure its fields into the local scope.
-    debug!(?remote_msg, "got message from remote peer");
+    // Wait for next message if the one we got is not Version
+    loop {
+        match remote_msg {
+            Message::Version { .. } => {
+                debug!(?remote_msg, "got version message from remote peer");
+                break;
+            }
+            _ => {
+                remote_msg = peer_conn
+                    .next()
+                    .await
+                    .ok_or(HandshakeError::ConnectionClosed)??;
+                debug!("ignoring non-version message from remote peer");
```

Comment:
Minor Logging: Can we log the message here, so we can see it when we are debugging connections?

```suggestion
                debug!(?remote_msg, "ignoring non-version message from remote peer");
```